src/main/java/io/confluent/connect/hdfs/HdfsSinkConnectorConfig.java

```diff
@@ -578,29 +586,26 @@ private static void addAllConfigKeys(ConfigDef container, ConfigDef other, Set<S
   }
 
   /**
-   * Extract the biggest integer contained in the string
-   *
-   * @param string - the string to extract the maximum integer from
-   * @return int - the biggest integer found in the string, 0 if none found
+   * Finds all instances of `${[0-9]+}` in `topics.dir` and returns a list.
+   * @return list of all numbers found in the regex pattern
    */
-  private static int extractMaximumInt(String string) {
-    int num = 0;
-    int res = 0;
+  private List<Integer> getPartsToReplace() {
 
-    for (char c : string.toCharArray()) {
+    List<Integer> toReplace = new ArrayList<>();
+    Pattern pattern = Pattern.compile("\\$\\{\\d+}");
```

Comment:
Do you need to escape the end bracket? Looks weird with mismatched escapes :-/

Reply:
its redundant to escape the second bracket - i can but i get a warning